blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/ShardStateManager.java

```diff
@@ -222,61 +235,56 @@ protected void updateSlotOnRead(SlotState slotState) {
             final int slot = slotState.getSlot();
             final long timestamp = slotState.getTimestamp();
             UpdateStamp.State state = slotState.getState();
+
+            //For slots in state "A", this would be last ingest time
+            //For slots in state "X", this would be last rollup time
             final long lastUpdateTimestamp = slotState.getLastUpdatedTimestamp();
 
-            UpdateStamp stamp = slotToUpdateStampMap.get(slot);
-            long nowMillis = new DateTime().getMillis();
-            if (stamp == null) {
+            UpdateStamp stampInMemory = slotToUpdateStampMap.get(slot);
+            if (stampInMemory == null) {
                 // haven't seen this slot before, take the update. This happens when a blueflood service is just started.
-                slotToUpdateStampMap.put(slot, new UpdateStamp(timestamp, state, false));
-            } else if (stamp.getTimestamp() != timestamp && state.equals(UpdateStamp.State.Active)) {
```

Comment:
I have questions about this section.  Just making a note so we don't forget to cover this.  Probably easiest to talk.